### DISCUSSION TITLE
How to train a pretrained R(2+1)D

### DISCUSSION THREAD

**Discussion by abhishek0696** [open-mmlab/mmaction2#1420] [Design]
Most of the models such as TANet, TPN, Timesformer has default code for pre-trained models (usually Imagenet). I want to train R(2+1)D on pretrained weights on imagenet or something.
But when I tried to change the pretrained thing in the base file like this:

![image](https://user-images.githubusercontent.com/63160543/152691486-c25b530a-4aef-4f5b-807e-e6d2e6402a1c.png)

It gave me this error: 

![image](https://user-images.githubusercontent.com/63160543/152691519-a7740472-54ee-4f86-8749-14ad75dfb7d6.png)

Please guide me on how to train the R(2+1)D on pretrained weights. Thanks!

**Comment by dreamerlin** [comment]
MMAction2 does not support train R(2+1)D on ImageNet weights since the shape of Conv2plus1d is not the same as that in ImageNet weight. If you still need to do this, maybe you can inflate the pretrained weight into R21D's shape in the way like I3D https://github.com/open-mmlab/mmaction2/blob/master/mmaction/models/backbones/resnet3d.py#L690.

If you want to finetune on your own dataset, you can use the K400 R21D weight in our modelzoo by changing `load_from`
